src/io/config.cpp

```diff
@@ -215,12 +216,15 @@ void Config::CheckParamConflict() {
     }
   }
   for (std::string metric_type : metric) {
+    bool metric_custom = metric_type == std::string("none") || metric_type == std::string("null") 
+                         || metric_type == std::string("custom") || metric_type == std::string("na");
     bool metric_type_multiclass = (CheckMultiClassObjective(metric_type) 
-                                    || metric_type == std::string("multi_logloss")
-                                    || metric_type == std::string("multi_error"));
+                                   || metric_type == std::string("multi_logloss")
+                                   || metric_type == std::string("multi_error")
+                                   || metric_custom);
     if ((objective_type_multiclass && !metric_type_multiclass)
       || (!objective_type_multiclass && metric_type_multiclass)) {
```

Comment:
Indent.

I'm doubt about the following situation:
`(!objective_type_multiclass && metric_type_multiclass)` : let say `objective=l2 && metric_type="None"`. Seems that it will trigger fatal error.

Reply:
yeah, good point.

Reply:
fixed